windtunnel/windtunnel.py

```diff
@@ -43,25 +43,99 @@
+# for the default wind tunnel size
+MAX_OBJECT_LENGTH_FACTOR = 0.5
+MAX_OBJECT_WIDTH_FACTOR = 0.2
+MAX_OBJECT_HEIGHT_FACTOR = 0.125
+
 
 class WindTunnel:
     """WindTunnel scenario."""
 
-    def __init__(self, inputs_dir: str = "./inductiva_input/"):
-        """Initializes the `WindTunnel` scenario object.
+    def __init__(self, dimensions: tuple[int, int, int] = (20, 10, 8)):
+        """
+        Initializes the wind tunnel with given dimensions.
+        length, width, height = dimensions
+
+        Parameters:
+        - dimensions: A tuple (length, width, height) for the tunnel size.
         """
+        self.length, self.width, self.height = dimensions
         self._walls = {
-            "x_min": -6,
-            "x_max": 14,
-            "y_min": -5,
-            "y_max": 5,
-            "z_min": 0,
-            "z_max": 8
+            "x_min": self.length * X_MIN_FACTOR,
+            "x_max": self.length * X_MAX_FACTOR,
+            "y_min": self.width * Y_MIN_FACTOR,
+            "y_max": self.width * Y_MAX_FACTOR,
+            "z_min": self.height * Z_MIN_FACTOR,
+            "z_max": self.height * Z_MAX_FACTOR,
+        }
+        self.dimensions = dimensions
+        self.object = None
+        self.object_area = 0
+        self.object_length = 0
+
+    def set_object(self,
+                   object_path: str,
+                   rotate_z_degrees: float = 0,
+                   normalize: bool = True,
+                   center: bool = True):
+        """Load an object into the windtunnel scenario. Optionally rotates, 
+        normalizes, and centers it.
+
+        Args:
+            object_path (str): Mesh file path.
+            rotate_z_degrees (float): Rotation around Z-axis in degrees.
+            normalize (bool): Scales object to unit size if True.
+            center (bool): Centers object in simulation space if True.
+        """
+
+        mesh = pv.read(object_path)
+        displace_vector = [0, 0, 0]
+        scaling_factor = 1
+
+        if center:
+            mesh, displace_vector, = pre_processing.move_mesh_to_origin(mesh)
+        if rotate_z_degrees:
+            mesh = mesh.rotate_z(rotate_z_degrees)
+        if normalize:
+            max_object_dimensions = (self.length * MAX_OBJECT_LENGTH_FACTOR,
+                                     self.width * MAX_OBJECT_WIDTH_FACTOR,
+                                     self.height * MAX_OBJECT_HEIGHT_FACTOR)
+            mesh, scaling_factor = pre_processing.normalize_mesh(
+                mesh, max_object_dimensions)
+
+        # Compute project area into the inlet face
+        self.object_area = pre_processing.compute_projected_area(
+            mesh, face_normal=(1, 0, 0))
+        self.object_length = pre_processing.compute_object_length(mesh)
+
+        self.object = mesh
+        return {
+            "displace_vector": displace_vector,
+            "scaling_factor": scaling_factor
```

Comment:
maybe return the rotate_z_degrees also on the dict

Reply:
```suggestion
            "rotate_z_degrees": rotate_z_degrees,
```